sdk/trace/simple_span_processor_test.go

```diff
@@ -40,9 +40,15 @@ func (t *testExporter) ExportSpans(ctx context.Context, spans []sdktrace.ReadOnl
 	return nil
 }
 
-func (t *testExporter) Shutdown(context.Context) error {
+func (t *testExporter) Shutdown(ctx context.Context) error {
 	t.shutdown = true
-	return nil
+	select {
+	case <-ctx.Done():
+		// Ensure context deadline tests receive the expected error.
+		return ctx.Err()
+	default:
+		return nil
+	}
```

Comment:
I think this is what made the tests stable 😉

Reply:
Yes it was. Based on the suggestion you made to reverse the error handling logic, this is required now.